avaframe/com7Regional/com7Regional.py

```diff
@@ -0,0 +1,456 @@
+
+        # Process peak files
+        peakFiles = list(avaDir.glob("Outputs/**/peakFiles/*.*"))
+        for peakFile in peakFiles:
+            targetPath = allPeakFilesDir / peakFile.name
+            operation(str(peakFile), str(targetPath))
+
+    return allPeakFilesDir
+
+
+def getRasterBounds(rasterFiles):
+    """Get the union bounds and validate cell sizes of multiple rasters.
+
+    Parameters
+    ----------
+    rasterFiles : list
+        List of paths to raster files
+
+    Returns
+    -------
+    bounds : dict
+        Dictionary containing xMin, yMin, xMax, yMax of the union bounds
+    cellSize : float
+        Cell size of the rasters
+
+    Raises
+    ------
+    ValueError
+        If cell sizes of rasters differ
+    """
+    # Read first raster to get cellSize and initialize bounds
+    firstRaster = rasterUtils.readRaster(rasterFiles[0])
+    cellSize = float(firstRaster["header"]["cellsize"])
+    bounds = {
+        "xMin": float("inf"),
+        "yMin": float("inf"),
+        "xMax": float("-inf"),
+        "yMax": float("-inf"),
+    }
+
+    # Find bounds and validate cell sizes
+    for rasterFile in rasterFiles:
+        raster = rasterUtils.readRaster(rasterFile)
+        header = raster["header"]
+
+        if float(header["cellsize"]) != cellSize:
+            raise ValueError(f"Different cell sizes found: {cellSize} vs {header['cellsize']}")
+
+        # Update bounds
+        bounds["xMin"] = min(bounds["xMin"], float(header["xllcenter"]))
+        bounds["yMin"] = min(bounds["yMin"], float(header["yllcenter"]))
+        bounds["xMax"] = max(
+            bounds["xMax"],
+            float(header["xllcenter"]) + float(header["ncols"]) * cellSize,
+        )
+        bounds["yMax"] = max(
+            bounds["yMax"],
+            float(header["yllcenter"]) + float(header["nrows"]) * cellSize,
+        )
+
+    return bounds, cellSize
+
+
+def mergeRasters(rasterFiles, bounds, mergeMethod="max"):
+    """Merge multiple rasters into a single raster.
+
+    Parameters
+    ----------
+    rasterFiles : list
+        List of paths to raster files
+    bounds : dict
+        Dictionary containing xMin, yMin, xMax, yMax of the union bounds
+    mergeMethod : str, optional
+        Method to use for merging overlapping cells. Options:
+        - 'max': maximum value (default)
+        - 'min': minimum value
+        - 'sum': sum of values
+        - 'count': number of overlapping valid results per cell
+
+    Returns
+    -------
+    mergedHeader : dict
+        Header dictionary containing ncols, nrows, xllcenter, yllcenter, cellsize, nodata_value
+    mergedData : numpy.ndarray
+        2D array containing the merged raster data
+    """
+
+    # Merge data with rasterio
+    # If something other than min/max is wanted, it is possible to provide a custom function to merge
+    mergedData, outputTransform = merge(rasterFiles, method=mergeMethod, masked=True)
```

Comment:
if the rasters have a different extent - does the rasterio merge function handle this?

Reply:
Yes, it could even handle different cellsizes...